src/main/java/net/kyori/text/BaseComponent.java

```diff
@@ -242,4 +201,107 @@ public String toString() {
 
   protected void populateToString(@Nonnull final MoreObjects.ToStringHelper builder) {
   }
+
+  /**
+   * An abstract implementation of a component builder.
+   *
+   * @param <B> the builder type
+   * @param <C> the component type
+   */
+  protected static abstract class AbstractBuilder<B extends AbstractBuilder<B, C>, C extends Component> implements Builder<B, C> {
+
+    /**
+     * The list of children.
+     *
+     * <p>This list is set to {@link #EMPTY_COMPONENT_LIST an empty list of components}
+     * by default to prevent unnecessary list creation for components with no children.</p>
+     */
+    @Nonnull protected List<Component> children = EMPTY_COMPONENT_LIST;
+    /**
+     * The color of this component.
+     */
+    @Nullable protected TextColor color;
+    /**
+     * If this component should have the {@link TextDecoration#OBFUSCATED obfuscated} decoration.
+     */
+    @Nonnull protected TextDecoration.State obfuscated = TextDecoration.State.NOT_SET;
```

Comment:
An `EnumMap<TextDecoration, TextDecoration.State>` might be more convenient for storing styles in case more get added in the future.